src/Compilers/Core/Portable/Syntax/SeparatedSyntaxList.cs

```diff
@@ -592,9 +592,16 @@ IEnumerator IEnumerable.GetEnumerator()
             return new SeparatedSyntaxList<SyntaxNode>(nodes._list);
         }
 
+        [Obsolete("Implicit downcast is not safe. Use CastDown<TNode>() instead", error: true)]
         public static implicit operator SeparatedSyntaxList<TNode>(SeparatedSyntaxList<SyntaxNode> nodes)
         {
             return new SeparatedSyntaxList<TNode>(nodes._list);
         }
+
+        public SeparatedSyntaxList<TDerived> CastDown<TDerived>()
+            where TDerived : TNode
+        {
+            return new SeparatedSyntaxList<TDerived>(_list);
```

Comment:
Note: this is still not safe.  Which is why we didn't go this way originally.

Reply:
This has been resolved by simply changing the implicit cast to an explicit cast.